### PR TITLE
added recursive container functions for learningRate and weightDecay

### PR DESCRIPTION
**Pull request by koltenpearson**
So you can set the learning rate and weight decay for the entire container at once.

### CODE REVIEW

**Comment by gpleiss** [comment]
Looks good! Thanks!
